(Bug) - #1503 Fix collectibles groups is not removed when the last item from it is sent

Agupane opened the pull request:
Closes #1503

gnosis-info commented:
Travis automatic deployment:
 https://pr1511--safereact.review.gnosisdev.com/app

mikheevm commented:
This looks more like a bug

mikheevm commented:
Closes how? What was the problem? Could you please add some description?

Agupane commented:
> Closes how? What was the problem? Could you please add some description?

The title got outdated, now I updated it, it solves issue #1503. It was intended to solve #1509 but found that the issue was already fixed while working on it, then I just fixed #1503 :)

fernandomg commented on src/logic/collectibles/store/selectors/index.ts:
```diff
@@ -16,11 +16,18 @@ export const nftAssetsListSelector = createSelector(nftAssets, (assets): NFTAsse
 })
 
+export const availableNftAssetsAddresses = createSelector(nftTokensSelector, (userNftTokens): string[] => {
+  return userNftTokens.map((nftToken) => nftToken.assetAddress)
```
won't it possibly return repeated addresses?

Agupane replied:
Yes you were right good catch! I fixed it

fernandomg commented on src/logic/collectibles/store/selectors/index.ts:
```diff
@@ -16,11 +16,18 @@ export const nftAssetsListSelector = createSelector(nftAssets, (assets): NFTAsse
+  (assets, activeAssetsList, availableNftAssetsAddresses): NFTAsset[] => {
+    return assets
+      .filter(({ address }) => activeAssetsList.has(address))
+      .filter((asset) => availableNftAssetsAddresses.includes(asset.address))
```
you could have used the same approach as the previous filter, `.filter(( { address }) =>` (my TOC talking here)

gnosis-info commented:
Travis automatic deployment:
 https://pr1511--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr1511--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr1511--safereact.review.gnosisdev.com/app

gnosis-info commented:
Travis automatic deployment:
 https://pr1511--safereact.review.gnosisdev.com/app

francovenica commented:
Looks good to me
Safe tested with: 

0xf8018825293f5724d671D8F6Dd961023ce0ba52d
0x9913B9180C20C6b0F21B6480c84422F6ebc4B808

Dev: 
![image.png](https://images.zenhubusercontent.com/5e14b8e3edee21d6f9aa6c8a/9431433c-8d55-4ab8-bfff-978941d6e3dd)

This PR: 
![image.png](https://images.zenhubusercontent.com/5e14b8e3edee21d6f9aa6c8a/3b6cfcc4-615e-4bd6-8064-428c264758da)
